include/mbedtls/arc4.h

```diff
@@ -19,10 +19,22 @@
  *  limitations under the License.
  *
  *  This file is part of mbed TLS (https://tls.mbed.org)
+ *
+ * \deprecated   ARC4 is considered a weak cipher and its use
+ *               constitutes a security risk. It is recommended
+ *               alternative ciphers should be considered instead.
+ *
  */
 #ifndef MBEDTLS_ARC4_H
 #define MBEDTLS_ARC4_H
 
+#if !defined(MBEDTLS_DEPRECATED_REMOVED)
+
+#if defined(MBEDTLS_DEPRECATED_WARNING)
+#warning ARC4 is deprecated and its use constitutes a security risk.
+#warning It is recommended alternative ciphers should be considered instead.
```

Comment:
The warning should mention that we're planning to remove it (in the same terms as the documentation). It's important practical information at this point: even developers who know/think that their use case is safe should feel concerned.

Reply:
Done.